stats/opentelemetry/client_metrics.go

```diff
@@ -156,6 +153,20 @@ func (h *clientMetricsHandler) TagConn(ctx context.Context, _ *stats.ConnTagInfo
 // HandleConn exists to satisfy stats.Handler.
 func (h *clientMetricsHandler) HandleConn(context.Context, stats.ConnStats) {}
 
+// getOrCreateRPCAttemptInfo retrieves or creates an rpc attemptInfo object
+// and ensures it is set in the context along with the rpcInfo.
+func getOrCreateRPCAttemptInfo(ctx context.Context) (context.Context, *attemptInfo) {
+	ri := getRPCInfo(ctx)
+	if ri == nil {
+		ri = &rpcInfo{}
+	}
+	ai := ri.ai
+	if ai == nil {
+		ai = &attemptInfo{}
+	}
+	return setRPCInfo(ctx, &rpcInfo{ai: ai}), ai
```

Comment:
This is adding it into the context a second time if it's already in there.

I think you want something more like:

```go
	ri := getRPCInfo(ctx)
	if ri != nil {
		return ctx, ri.ai
	}
	ri := &rpcInfo{ai: &attemptInfo{}}
	return setRPCInfo(ctx, ri), ri.ai
```